### PR TITLE
Authoritative NAF first-syncs

### PR DESCRIPTION
**Pull request by brianpeiris**
We want to securely associate "first sync" NAF messages with the associated socket's session id. To that end, we now generate the socket session id in reticulum, instead of allowing the client to tell us their own session id. We then use this session id to override the "creator" and "owner" fields in "first sync" networked-aframe messages.

### CODE REVIEW

File: lib/ret_web/channels/session_socket.ex [hunk 1]
```diff
@@ -12,10 +12,25 @@ defmodule RetWeb.SessionSocket do
     "session:#{socket.assigns.session_id}"
   end
 
-  def connect(%{"session_id" => session_id}, socket) do
+  def connect(%{"session_token" => session_token}, socket) do
+    session_id =
```
Comment (gfodor) [inline lib/ret_web/channels/session_socket.ex]: This seems worth a comment

File: lib/ret_web/channels/session_socket.ex [hunk 2]
```diff
@@ -12,10 +12,25 @@ defmodule RetWeb.SessionSocket do
...
+      case session_token |> Ret.SessionToken.decode_and_verify() do
```
Comment (gfodor) [inline lib/ret_web/channels/session_socket.ex]: I'd push this little routine into a function on `SessionToken` so you can then just do:
```
assign(:session_id, session_token |> SessionToken.session_id_for_token || SecureRandom.uuid()
```

also might be nice to extract `SecureRandom.uuid()` into a function `generate_session_id` for documentation purposes

File: lib/ret/session_token.ex [hunk 3]
```diff
@@ -0,0 +1,35 @@
+defmodule Ret.SessionToken do
```
Comment (gfodor) [inline lib/ret/session_token.ex]: We have a few of these JWT tokens now, I think we should add a one or two line comment at the top of all these describing them so we can keep them straight

File: lib/ret_web/channels/session_socket.ex (hunk 1 above)
Comment (brianpeiris) [reply lib/ret_web/channels/session_socket.ex]: Hmm, not sure what you're getting at. The session token allows us to trust the session id contained in the token. Should I add a comment to the code saying that?

Comment (brianpeiris) [reply lib/ret_web/channels/session_socket.ex]: Perhaps the moduledoc I just added is sufficient?

**Pull request merged**